src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1733,7 +1733,7 @@ private void HandleHostError(Exception exception)
             TraceWriter.Error(message, exception);
             TraceWriter.Flush();
 
-            Logger?.LogError(0, exception, message);
+            // Note: We do not log to ILogger here as any error has already been logged.
```

Comment:
where is the other codepath which logs the exception to ILogger?

Reply:
It happens in the SDK when we log the function result. If it's an error, it logs that Exception. Then this subscribes to the TraceWriter's errors and logs it again. Right now I'm trying to figure out how to hit those other code paths to make sure they're logging to ILogger as well (if TraceWriter.Error() is being called, which is how this method runs, then it should be).

Reply:
Yeah, it logs one Exception in the other scenarios as expected now. Basically we have it wired up with TraceWriter such that somewhere you do this: `traceWriter.Error()` -> `HandleHostError()`, which makes sure that the log is put in the correct function-specific file. But everywhere we have `traceWriter.Error()`, we also have `ILogger.LogError()`. So putting it inside of `HandleHostError()` results in dupes. And since `ILogger` isn't responsible for any host-level file logging, there's nothing for it to do here.

Reply:
Ok, that makes sense, thanks for the explanation @brettsam!